extensions/client-batch.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "client-initiated batch Extension"
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Val Lorentz"
+    period: "2021"
+    email: "progval+ircv3@progval.net"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+All specifications depending on this specification MUST be draft specification
+and use only capability names prefixed by `draft/`
+
+## Introduction
+
+This specification extends the [`batch`][] specification to allow
+client-initiated BATCH commands.
+
+This specification itself does not introduce any client-to-server
+batch type, but is designed as a framework for other specifications.
+
+## Motivation
+
+Certain specifications call for clients to be able to send batched messages to
+servers, for instance [`draft/multiline`][]
+
+## Architecture
+
+### Dependencies
+
+This specification does not depend on the [`batch`][] capability,
+which is used for server-initiated batches only.
+
+This specification uses the [standard replies][] framework.
+
+### Capabilities
+
+This specification does not introduce any capability.
+
+Clients SHOULD NOT send `BATCH` commands to servers, unless they negotiated
+a capability that allows them to do so.
+
+### The client-to-server `BATCH` verb
+
+This specification introduces client initiated batches.
+These use the same syntax as server initiated batches.
+
+Once a client has opened a batch, it MUST NOT send any messages
+that are not part of the batch, until it is closed
+(with `BATCH -reference-tag`).
+
+Clients MUST NOT send nested batches, i.e. clients may not send `BATCH`
+commands with a `batch` tag.
+
+### Errors
+
+Servers MUST use `FAIL` messages from the [standard replies][] framework
+to notify clients of errors with client-initiated batches.
+The command is `BATCH` and the following codes are defined:
+
+* `TIMEOUT`: the batch was left open for too long, all future messages
+  in this batch will be ignored.
```

Comment:
Starting from which moment? As worded now, the client has no way of knowing which messages are ignored

Reply:
True. Should all messages of the batch be ignored?

Reply:
That's an option. However, it creates another corner case: server sends `FAIL TIMEOUT` just after client closes the batch. Now in the case of success client can't know if the that was a success; it will learn of a failure though.

Reply:
Clients should get something echoed back to them (eg. with `echo-message`) if the batch was successful. And I can add the batch id as argument of `FAIL TIMEOUT` to disambiguate the failure case

Reply:
Only for messages which do provide some feedback. But since it's a generic framework, the messages could be anything

Reply:
Do we currently have any type of message that doesn't provide feedback? If yes, they should probably be "fixed" outside this spec, because feedback is also important outside a batch.